frontend/packages/console-app/src/components/tour/tour-context.ts

```diff
@@ -0,0 +1,86 @@
+import { createContext, useReducer, Reducer, Dispatch, ReducerAction } from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import { getFlagsObject } from '@console/internal/reducers/features';
+import { useExtensions } from '@console/plugin-sdk';
+import { RootState } from '@console/internal/redux';
+import { isGuidedTour, GuidedTour } from '@console/plugin-sdk/src/typings/guided-tour';
+import { TourActions } from './const';
+import { getTourLocalStorageForPerspective, filterTourBasedonPermissionAndFlag } from './utils';
+import { TourDataType } from './type';
+
+export const tourReducer = (state: TourState, action: TourActions) => {
+  const { stepNumber } = state;
+  switch (action) {
+    case TourActions.start:
+      return { startTour: true, closeTour: false, completedTour: false, stepNumber: 0 };
+    case TourActions.next:
+      return {
+        startTour: false,
+        closeTour: false,
+        completedTour: false,
+        stepNumber: stepNumber + 1,
+      };
+    case TourActions.back:
+      return {
+        startTour: false,
+        closeTour: false,
+        completedTour: false,
+        stepNumber: stepNumber - 1,
+      };
+    case TourActions.pause:
+      return {
+        startTour: false,
+        closeTour: true,
+        stepNumber,
+        completedTour: false,
+      };
+    case TourActions.resume:
+      return { startTour: false, closeTour: false, stepNumber, completedTour: false };
+    case TourActions.complete:
+      return { startTour: false, completedTour: true, closeTour: false, stepNumber: 0 };
+    default:
+      return state;
+  }
+};
+
+type TourReducer = Reducer<TourState, TourActions>;
+
+type TourContextType = {
+  tourState?: TourState;
+  tourDispatch?: Dispatch<ReducerAction<TourReducer>>;
+  tour?: TourDataType;
+  totalSteps?: number;
+};
+
+export type TourState = {
+  stepNumber: number;
+  startTour: boolean;
+  closeTour: boolean;
+  completedTour: boolean;
+};
+
+export const TourContext = createContext<TourContextType>({});
+
+export const useTourValuesForContext = (): TourContextType => {
+  const activePerspective = useSelector(({ UI }: RootState): string => UI.get('activePerspective'));
```

Comment:
```suggestion
  const activePerspective = useSelector(getActivePerspective);
```

Reply:
import the getActivePerspective in this file is causing some issues for ContextProvider Extension.
Ends up with this error:
```
job.js:17 Uncaught ReferenceError: Cannot access 'allModels' before initialization
    at Module.allModels (job.js:17)
    at Module.<anonymous> (ui.ts:106)
    at Module../public/actions/ui.ts (ui.ts:371)
    at __webpack_require__ (bootstrap:878)
    at fn (bootstrap:150)
    at Module.<anonymous> (ui.ts:1)
    at Module../public/reducers/ui.ts (ui.ts:395)
    at __webpack_require__ (bootstrap:878)
    at fn (bootstrap:150)
    at Module.<anonymous> (StepHeader.tsx:10)
```

Reply:
This must be because of how the reducer is not lazy loaded but loaded immediately upfront with the extensions :(

Please add a comment as to why you are doing what you are doing then as a workaround.

New declarative extensions mechanism will support this.

Reply:
added comment